### ISSUE TITLE
Very slow on Tablets

### ISSUE THREAD

**Issue by dustinduse** [sengirab/ngAutocomplete#23]
I am having an issue where this is very slow on tablets. It causes the page to hang on loading for 10-15 seconds then when you try to open the box it can take 5 seconds before anything happens.

Has anyone had this issue before?. Or is this not designed for such large amounts of options? I have 1409.

**Comment by sengirab** [comment]
Are you experiencing this only on tablet?

I tested my dropdown with 1000. Performance was OK. I’ll look into the performance once more.

**Comment by dustinduse** [comment]
Yes. On a pc there is no real noticeable hang or lag to open the drop down. Is it possible to render the box serverside?

**Comment by sengirab** [comment]
I dont think server rendering will fix your lag to be honest.. Its a lightweight angular component. It wont really add much to your bundle size, slow internet could be a cause of your initial load time.

Anyway, https://angular.io/guide/universal might be worth checking out if we’re talking about server side rendering.

Another option would be to make your dropdown async, e.g creating an endpoint to fetch results filling the dropdown. Its explained in #17

**Comment by dustinduse** [comment]
Yes, I toyed with that a little bit but was unable to make it work properly.
